experimental/packages/sdk-logs/src/LoggerProvider.ts

```diff
@@ -0,0 +1,164 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { diag } from '@opentelemetry/api';
+import type * as logsAPI from '@opentelemetry/api-logs';
+import { IResource, Resource } from '@opentelemetry/resources';
+import { getEnv, merge } from '@opentelemetry/core';
+
+import type { LoggerProviderConfig, LogRecordLimits } from './types';
+import type { LogRecordProcessor } from './LogRecordProcessor';
+import type { LogRecordExporter } from './export/LogRecordExporter';
+import { Logger } from './Logger';
+import { loadDefaultConfig } from './config';
+import { MultiLogRecordProcessor } from './MultiLogRecordProcessor';
+import { BatchLogRecordProcessor } from './platform/node/export/BatchLogRecordProcessor';
+import { NoopLogRecordProcessor } from './export/NoopLogRecordProcessor';
+
+export type EXPORTER_FACTORY = () => LogRecordExporter;
+
+export class LoggerProvider implements logsAPI.LoggerProvider {
+  protected static readonly _registeredExporters = new Map<
+    string,
+    EXPORTER_FACTORY
+  >();
+
+  public readonly resource: IResource;
+
+  private readonly _loggers: Map<string, Logger> = new Map();
+  private readonly _logRecordLimits: LogRecordLimits;
+  private _activeProcessor: MultiLogRecordProcessor;
+  private readonly _registeredLogRecordProcessors: LogRecordProcessor[] = [];
+  private readonly _forceFlushTimeoutMillis;
+
+  constructor(config: LoggerProviderConfig = {}) {
+    const { resource, logRecordLimits, forceFlushTimeoutMillis } = merge(
+      {},
+      loadDefaultConfig(),
+      config
+    );
+    this.resource = Resource.default().merge(resource ?? Resource.empty());
+    this._logRecordLimits = logRecordLimits;
+    this._forceFlushTimeoutMillis = forceFlushTimeoutMillis;
+
+    const defaultExporter = this._buildExporterFromEnv();
+    if (defaultExporter !== undefined) {
+      this._activeProcessor = new MultiLogRecordProcessor(
+        [new BatchLogRecordProcessor(defaultExporter)],
+        forceFlushTimeoutMillis
+      );
+    } else {
+      this._activeProcessor = new MultiLogRecordProcessor(
+        [new NoopLogRecordProcessor()],
+        forceFlushTimeoutMillis
+      );
+    }
+  }
+
+  /**
+   * Get a logger with the configuration of the LoggerProvider.
+   */
+  public getLogger(
+    name: string,
+    version?: string,
+    options?: logsAPI.LoggerOptions
+  ): Logger {
+    const { schemaUrl = '' } = options || {};
+    const key = `${name}@${version || ''}:${schemaUrl}`;
+    if (!this._loggers.has(key)) {
+      this._loggers.set(
+        key,
+        new Logger({
+          resource: this.resource,
+          logRecordLimits: this._logRecordLimits,
+          activeProcessor: this._activeProcessor,
```

Comment:
There is a [requirement](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#logger-creation) that updates to the provider configuration is reflected automatically on existing logger instances. I think passing the processor in this way would not reflect an update to a processor.